R/methods.R

```diff
@@ -101,10 +101,14 @@ print.lints <- function(x, ...) {
     if (isTRUE(settings$error_on_lint)) {
       quit("no", 31L, FALSE) # nocov
     }
-  } else if (use_rstudio_source_markers) {
+  }
+
+  if (length(x) == 0L) {
```

Comment:
Extracting this branch to a separate conditional because the code is already quite nested.

Reply:
This should be an `else` IMO because `!(length(x) > 0)` implies `length(x) == 0`.

Reply:
Yes, but I am trying to reduce the nesting here, and it is easy to do so here by separating logic relevant for empty versus non-empty lint cases. 

Otherwise, the code is going to look like this:

```r
  if (length(x) > 0L) {
    inline_data <- x[[1L]][["filename"]] == "<text>"
    if (!inline_data && use_rstudio_source_markers) {
      rstudio_source_markers(x)
    } else if (in_github_actions() && !in_pkgdown()) {
      github_actions_log_lints(x, project_dir = github_annotation_project_dir)
    } else {
      lapply(x, print, ...)
    }

    if (isTRUE(settings$error_on_lint)) {
      quit("no", 31L, FALSE) # nocov
    }
  } else {
    cli_inform(c(i = "No lints found."))
    # Empty lints: clear RStudio source markers
    if (use_rstudio_source_markers) { 
      rstudio_source_markers(x)
    }
  }
```

Instead of 

```r
  if (length(x) > 0L) {
    inline_data <- x[[1L]][["filename"]] == "<text>"
    if (!inline_data && use_rstudio_source_markers) {
      rstudio_source_markers(x)
    } else if (in_github_actions() && !in_pkgdown()) {
      github_actions_log_lints(x, project_dir = github_annotation_project_dir)
    } else {
      lapply(x, print, ...)
    }

    if (isTRUE(settings$error_on_lint)) {
      quit("no", 31L, FALSE) # nocov
    }
  }

  if (length(x) == 0L) {
    cli_inform(c("i" = "No lints found."))
    # Empty lints: clear RStudio source markers
    if (use_rstudio_source_markers) rstudio_source_markers(x)
  }
```

Reply:
I see no issue with the former. Maybe add a comment to the else, explaining that this handles the empty case?